packages/did-provider-jwk/src/resolver.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  DIDDocument,
+  DIDResolutionOptions,
+  DIDResolutionResult,
+  DIDResolver,
+  ParsedDID,
+  Resolvable,
+  JsonWebKey,
+} from 'did-resolver'
+import { encodeBase64url, decodeBase64url } from '@veramo/utils'
+
+const isJWK = (data: unknown): data is JsonWebKey => {
+  if (
+    typeof data === 'object' &&
+    data &&
+    'crv' in data &&
+    typeof data.crv === 'string' &&
+    'kty' in data &&
+    'x' in data &&
+    typeof data.x === 'string' &&
+    ((data.kty === 'EC' && 'y' in data && typeof data.y === 'string') ||
+      (data.kty === 'OKP' && !('y' in data)))
+  ) {
+    return true
+  }
+  return false
+}
+
+function generateDidDocument(jwk: JsonWebKey): Promise<DIDDocument> {
+  return new Promise((resolve, reject) => {
+    try {
+      const did = `did:jwk:${encodeBase64url(JSON.stringify(jwk))}`
+      const didDocument: DIDDocument = {
+        id: did,
+        '@context': ['https://www.w3.org/ns/did/v1', 'https://w3id.org/security/suites/jws-2020/v1'],
+        verificationMethod: [
+          {
+            id: `${did}#0`,
+            type: 'JsonWebKey2020',
+            controller: did,
+            publicKeyJwk: jwk,
+          },
+        ],
+        assertionMethod: [`${did}#0`],
+        authentication: [`${did}#0`],
+        capabilityInvocation: [`${did}#0`],
+        capabilityDelegation: [`${did}#0`],
+        keyAgreement: [`${did}#0`],
+      }
+      resolve(didDocument)
+    } catch (error) {
+      reject(error)
+    }
+  })
+}
+
+function parseDidJwkIdentifier(didIdentifier: string): JsonWebKey {
+  try {
+    const jwk = JSON.parse(decodeBase64url(didIdentifier)) as unknown
+    if (!isJWK(jwk)) {
+      throw new Error("DID identifier doesn't contain a valid JWK")

```

Comment:
We try to prefix error messages for easier interpretation.
In the case of this resolver, all errors fall under an `illegal_argument: ` prefix

```suggestion
      throw new Error("illegal_argument: DID identifier doesn't contain a valid JWK")
```